File: Week 2 - Problems/readability.c
#include <cs50.h>
#include <ctype.h>
#include <math.h>
#include <stdio.h>
#include <string.h>

float count_grade(string text);

int main(void)
{
    string text = get_string("Text: ");
    float grade = count_grade(text);
    grade = round(grade);

    if (grade < 1)
    {
        printf("Before Grade 1\n");
    }
    else if (grade >= 16)
    {
        printf("Grade 16+\n");
    }
    else
    {
        printf("Grade %0.0f\n", grade);
    }

    return 0;
}

float count_grade(string text)
{
    float number_words = 1;
    int number_letters = 0;
    int number_sentences = 0;

    for (int i = 0; i < strlen(text); i++)
    {
        if ((text[i] >= 'A' && text[i] <= 'Z') || (text[i] >= 'a' && text[i] <= 'z'))
        {
            number_letters += 1;
        }
        else if (text[i] == '.' || text[i] == '!' || text[i] == '?')
        {
            number_sentences += 1;
        }
        else if (text[i] == ' ')
        {
            number_words += 1;
        }
    }

    float L = (number_letters / number_words) * 100;
    float S = (number_sentences / number_words) * 100;

    float grade = 0.0588 * L - 0.296 * S - 15.8;

    return grade;
}

File: Week 6 - Problems/readability.py

def count_grade(text):
    number_words = 1
    number_letters = 0
    number_sentences = 0

    for i in range(len(text)):
        if ((text[i] >= 'A' and text[i] <= 'Z') or (text[i] >= 'a' and text[i] <= 'z')):
            number_letters += 1
        elif (text[i] == '.' or text[i] == '!' or text[i] == '?'):
            number_sentences += 1
        elif (text[i] == ' '):
            number_words += 1

    L = (number_letters / number_words) * 100
    S = (number_sentences / number_words) * 100

    grade = 0.0588 * L - 0.296 * S - 15.8

    return grade


text = input("Text: ")
grade = count_grade(text)
grade = round(grade)

if (grade < 1):
    print("Before Grade 1")
elif (grade >= 16):
    print("Grade 16+")
else:
    print(f"Grade {grade}")
